engine/src/main/java/org/camunda/bpm/engine/impl/history/producer/DefaultDmnHistoryEventProducer.java

```diff
@@ -101,8 +101,14 @@ protected HistoryEvent createHistoryEvent(DmnDecisionEvaluationEvent evaluationE
     HistoricDecisionEvaluationEvent event = newDecisionEvaluationEvent(evaluationEvent);
 
     HistoricDecisionInstanceEntity rootDecisionEvent = supplier.createHistoricDecisionInstance(evaluationEvent.getDecisionResult(), null);
+    /**
+     * Setting the generated decisionInstanceId to the root event
+     * Generating custom decisionInstanceId to persist the history event and return the id in the response
+     */
+    if(isNotBlank(evaluationEvent.getDecisionInstanceId())) {
+      rootDecisionEvent.setId(evaluationEvent.getDecisionInstanceId());
```

Comment:
Note to myself:
❓ Shouldn't the id of the root to be pointing to the root event DI id? What's the reference?